<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1500</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1500</width>
    <height>800</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1500</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1501</width>
      <height>1252</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_bldcConf">
     <attribute name="title">
      <string>Debug tab</string>
     </attribute>
     <widget class="PlotWidget" name="graphWidget" native="true">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>0</y>
        <width>1351</width>
        <height>541</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pb_start">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>710</x>
        <y>670</y>
        <width>161</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Start graph</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pb_resetMotor">
      <property name="geometry">
       <rect>
        <x>890</x>
        <y>670</y>
        <width>81</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset Motor</string>
      </property>
     </widget>
     <widget class="QComboBox" name="cb_controlType">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>570</y>
        <width>99</width>
        <height>30</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>70</width>
        <height>30</height>
       </size>
      </property>
      <item>
       <property name="text">
        <string>Current control</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Speed control</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Position control</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="lb_boardStatus">
      <property name="geometry">
       <rect>
        <x>710</x>
        <y>730</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Board offline</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>240</y>
        <width>102</width>
        <height>226</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_19">
       <item>
        <widget class="QCheckBox" name="chb_currentIq">
         <property name="text">
          <string>Current Iq</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_currentId">
         <property name="text">
          <string>Current Id</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_desiredIq">
         <property name="text">
          <string>Desired Iq</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_currentIab">
         <property name="text">
          <string>Current Iab</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_currentUdq">
         <property name="text">
          <string>Current Udq</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_currentUab">
         <property name="text">
          <string>Current Uab</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_currentPos">
         <property name="text">
          <string>Current position</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_desiredPos">
         <property name="text">
          <string>Desired position</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_currentSpeed">
         <property name="text">
          <string>Current speed</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_desiredSpeed">
         <property name="text">
          <string>Desired speed</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>121</width>
        <height>103</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QPushButton" name="pb_searchPort">
           <property name="text">
            <string>Search ports</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pb_connect">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QComboBox" name="cb_ports"/>
         </item>
         <item>
          <widget class="QLabel" name="lb_status">
           <property name="text">
            <string>Disconnected</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>98</width>
        <height>111</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Data to acquire</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_rawdata">
         <property name="text">
          <string>Raw data</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QCheckBox" name="chb_iacurr">
           <property name="text">
            <string>Current Ia</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="chb_ibcurr">
           <property name="text">
            <string>Current Ib</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="chb_iccurr">
           <property name="text">
            <string>Current Ic</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>470</y>
        <width>101</width>
        <height>71</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_21">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_20">
         <item>
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>Plot points</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="le_plotPoints">
           <property name="text">
            <string>2000</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="pb_clearData">
         <property name="text">
          <string>Clear Data</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>610</y>
        <width>158</width>
        <height>42</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QPushButton" name="pb_readConf">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>Read config</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pb_writeConf">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string>Write config</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget_2">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>650</y>
        <width>161</width>
        <height>97</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_10">
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Current Q</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_11">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>KP</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>KI</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>KD</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <layout class="QVBoxLayout" name="verticalLayout_12">
           <item>
            <widget class="QSpinBox" name="sb_currQkP">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_currQkI">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_currQkD">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget_3">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>650</y>
        <width>161</width>
        <height>97</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_13">
       <item>
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Speed</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_3">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_14">
           <item>
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>KP</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>KI</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>KD</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <layout class="QVBoxLayout" name="verticalLayout_15">
           <item>
            <widget class="QSpinBox" name="sb_speedkP">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_speedkI">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_speedkD">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget_4">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>650</y>
        <width>161</width>
        <height>97</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_16">
       <item>
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Position</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_4">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_17">
           <item>
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>KP</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>KI</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_17">
             <property name="text">
              <string>KD</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <layout class="QVBoxLayout" name="verticalLayout_18">
           <item>
            <widget class="QSpinBox" name="sb_poskP">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_poskI">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_poskD">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QPushButton" name="pb_runMotor">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>1000</x>
        <y>670</y>
        <width>81</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Run Motor</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="cb_sendData">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>560</y>
        <width>70</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Send data</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pb_saveData">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>540</y>
        <width>101</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Save data</string>
      </property>
     </widget>
     <widget class="QWidget" name="">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>651</y>
        <width>161</width>
        <height>97</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Current D</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>KP</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>KI</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>KD</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QSpinBox" name="sb_currDkP">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_currDkI">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_currDkD">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>560</y>
        <width>1081</width>
        <height>84</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_23">
         <item>
          <widget class="QLabel" name="label_22">
           <property name="text">
            <string>Speed</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_23">
           <property name="text">
            <string>Curr Q</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_24">
           <property name="text">
            <string>Speed</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QSlider" name="hs_speedVal">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="maximum">
              <number>1600</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="hs_currQval">
             <property name="minimum">
              <number>-65535</number>
             </property>
             <property name="maximum">
              <number>65535</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_22">
           <item>
            <widget class="QLabel" name="label_19">
             <property name="text">
              <string>TextLabel</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_currQval">
             <property name="minimum">
              <number>-65535</number>
             </property>
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_2"/>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Tab 2</string>
     </attribute>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1500</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave_Config"/>
    <addaction name="actionLoad_config"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionSave_Config">
   <property name="text">
    <string>Save Config</string>
   </property>
  </action>
  <action name="actionLoad_config">
   <property name="text">
    <string>Load config</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>hs_speedVal</sender>
   <signal>valueChanged(int)</signal>
   <receiver>label_19</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>984</x>
     <y>616</y>
    </hint>
    <hint type="destinationlabel">
     <x>1135</x>
     <y>618</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_currQval</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sb_currQval</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1108</x>
     <y>644</y>
    </hint>
    <hint type="destinationlabel">
     <x>1130</x>
     <y>649</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sb_currQval</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_currQval</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1138</x>
     <y>641</y>
    </hint>
    <hint type="destinationlabel">
     <x>1108</x>
     <y>640</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
